sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/_tests_/assessment-workflow-enqueuer.scheduler.e2e-spec.ts

```diff
@@ -186,6 +195,74 @@ describe(
       expect(startApplicationAssessmentQueueMock.add).not.toBeCalled();
     });
 
+    it(`Should queue the assessment cancellation when an application has one assessment with status '${StudentAssessmentStatus.CancellationRequested}'.`, async () => {
+      // Arrange
+
+      // Application with a cancellation requested assessment.
+      const application = await createDefaultApplication(
+        StudentAssessmentStatus.CancellationRequested,
+      );
+      application.applicationStatus = ApplicationStatus.Overwritten;
+      application.studentAssessments = [application.currentAssessment];
+      await db.application.save(application);
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Act
+      await processor.enqueueAssessmentOperations(job);
+
+      // Assert
+      // Load application and related assessments for assertion.
+      const updatedApplication = await loadApplicationDataForAssertions(
+        application.id,
+      );
+
+      // Assert item was added to the queue.
+      const queueData = {
+        assessmentId: updatedApplication.currentAssessment.id,
+      } as CancelAssessmentQueueInDTO;
+      expect(cancelApplicationAssessmentQueueMock.add).toBeCalledWith(
+        queueData,
+      );
+
+      expect(updatedApplication.currentAssessment.studentAssessmentStatus).toBe(
+        StudentAssessmentStatus.CancellationQueued,
+      );
+    });
+
+    it(`Should not queue the assessment cancellation when an application has one assessment with status '${StudentAssessmentStatus.CancellationRequested}' but has another assessment with status '${StudentAssessmentStatus.CancellationQueued}'.`, async () => {
+      // Arrange
+
+      // Application with a cancellation queued assessment.
+      const application = await createDefaultApplication(
+        StudentAssessmentStatus.CancellationQueued,
+      );
+      const submittedAssessment = createFakeStudentAssessment({
+        application,
+        auditUser,
+      });
+      submittedAssessment.studentAssessmentStatus =
+        StudentAssessmentStatus.CancellationRequested;
+      await db.studentAssessment.save(submittedAssessment);
+
+      application.applicationStatus = ApplicationStatus.Overwritten;
+      application.studentAssessments = [
+        application.currentAssessment,
+        submittedAssessment,
+      ];
+      await db.application.save(application);
```

Comment:
Maybe I am missing something but it the above code not equivalent to the below one?
I am not following, for instance, why the line 241 creates an assessment `submittedAssessment` and then it is updated to `CancellationRequested`?

```ts
// Application with a cancellation queued assessment.
const application = await createDefaultApplication(
  StudentAssessmentStatus.CancellationQueued,
);
const cancellationRequestedAssessment = createFakeStudentAssessment({
  auditUser,
  application,
});
await db.studentAssessment.save(cancellationRequestedAssessment);
```

Reply:
The idea for the test is to have an application with an assessment with `CancellationQueued` status and another one with `CancellationRequested` and assert that this assessment with `CancellationRequested` status won't be queued.
The code above generates an application with `CancellationQueued` and `Submitted` statuses.